lib/saluki-components/src/sources/checks/python_exposed_modules.rs

```diff
@@ -0,0 +1,162 @@
+use tracing::trace;
+
+use super::*;
+
+/// submit_metric is called from the AgentCheck implementation when a check submits a metric.
+/// Python signature:
+///     aggregator.submit_metric(self, self.check_id, mtype, name, value, tags, hostname, flush_first_value)
+///
+/// TODO(remy): should mtype be a PyMetricType?
+#[allow(clippy::too_many_arguments)]
+#[pyfunction]
+#[pyo3(pass_module)]
+pub(crate) fn submit_metric(
+    module: &Bound<PyModule>, _class: PyObject, _check_id: String, mtype: i32, name: String, value: f64,
+    tags: Vec<String>, hostname: String, _flush_first_value: bool,
+) {
+    trace!(
+        "submit_metric called with name: {}, value: {}, tags: {:?}, hostname: {}",
+        name,
+        value,
+        tags,
+        hostname
+    );
+
+    let check_metric = CheckMetric {
+        name,
+        metric_type: mtype.into(),
+        value,
+        tags,
+    };
+
+    match module.getattr("SUBMISSION_QUEUE") {
+        Ok(py_item) => match py_item.extract::<Py<python_scheduler::PythonSenderHolder>>() {
+            Ok(q) => {
+                let res = pyo3::Python::with_gil(|py| q.bind_borrowed(py).borrow_mut().sender.clone());
```

Comment:
Nit, I don't think the gil needs to be acquired here - see https://pyo3.rs/main/performance#access-to-bound-implies-access-to-gil-token

Reply:
ah, you're right, that lets me get rid of this `sender.clone` as well, good spot!